src/components/MediaInput/MediaInput.test.tsx

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import { render, waitFor } from '@testing-library/react';
+import user from '@testing-library/user-event';
+import MediaInput, { MediaInputProps } from './MediaInput';
+
+const renderComponent = (props: MediaInputProps) => render(<MediaInput {...props} />);
+
+const initialProps: MediaInputProps = {
+  mediaType: '',
+  previewUrl: '',
+  hasError: false,
+  onChange: jest.fn()
+};
+
+test('should not render input element', () => {
+  const { getByText } = renderComponent(initialProps);
+  const input = getByText('Choose Media') as HTMLInputElement;
+  expect(input).toBeVisible();
+});
+
+test('should render choose media text when previewUrl empty', () => {
+  const { getByText } = renderComponent(initialProps);
+  const input = getByText('Choose Media') as HTMLInputElement;
+  expect(input).toBeVisible();
+});
+
+test('should render preview image when previewUrl valid and mediaType equals image', () => {
+  const { container } = renderComponent({
+    ...initialProps,
+    mediaType: 'image',
+    previewUrl: 'testing.png'
+  });
+
+  expect(container.querySelector('img')).toBeVisible();
```

Comment:
Can we also assert that the `img` `src` is `previewUrl` (same for the video one below)?